java/ql/src/experimental/Security/CWE/CWE-1104/UsesInsecureDependency.ql

```diff
@@ -0,0 +1,23 @@
+/**
+ * @name Using Insecure Dependency 
+ * @description A dependency imported into your project is either deprecated or will cease to be supported in the near future.
+ * @kind problem
+ * @problem.severity error
+ * @precision high
+ * @id java/microsoft/using-insecure-dependency
+ * @tags security
+ *       external/cwe/cwe-1104
+ */
+import java
+
+
+predicate usesInsecureDependency(Import i) {
+  i.toString().splitAt(" ", 1) in [ "com.microsoft.aad.adal4j.*", 
+    							    "AuthenticationContext",
+    								"X509Certificate"
+  								  ]
+}
+
+from Import imp
+where imp = max(Import i | usesInsecureDependency(i) | i order by i.getFile().toString())
```

Comment:
This would only give one alert even if they used two different deprecated packages. How about writing a predicate that takes the max Import *per dependency*, so that if I have imports

```
import deprecated1.SomeClass1
import deprecated2.SomeClass2
import deprecated1.SomeOtherClass1
import deprecated2.SomeOtherClass2
import deprecated1.SomeOtherClass1
```

Then I should get two alerts: use of dependency `deprecated1`, use of dependency `deprecated2`.